src/spikeinterface/sortingcomponents/peak_localization.py

```diff
@@ -289,17 +289,22 @@ class LocalizeGridConvolution(PipelineNode):
         The margin for the grid of fake templates
     prototype: np.array
         Fake waveforms for the templates. If None, generated as Gaussian
+    percentile: float (default 10)
+        The percentage in [0, 100] of the best scalar products kept to 
+        estimate the position
     """
     def __init__(self, recording, return_output=True, parents=['extract_waveforms'], local_radius_um=50., upsampling_um=5,
-        sigma_um=np.linspace(10, 50., 5), sigma_ms=0.25, margin_um=50., prototype=None):
+        sigma_um=np.linspace(10, 50., 5), sigma_ms=0.25, margin_um=50., prototype=None, percentile=10):
         PipelineNode.__init__(self, recording, return_output=return_output, parents=parents)
         
         self.local_radius_um = local_radius_um
         self.sigma_um = sigma_um
         self.margin_um = margin_um
         self.upsampling_um = upsampling_um
-        contact_locations = recording.get_channel_locations()
+        self.percentile = 100 - percentile
+        assert 0 <= self.percentile <= 100, "Percentile should be in [0, 100]"
```

Comment:
Make sure the mask keeps at least some dot products

Reply:
I am not worried about that and/or I do not get the point. We are generated hundreds of templates, and thus even a percentile of 1 will ensure enough non_zeros dot products